packages/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts

```diff
@@ -214,6 +216,61 @@ export default class MongoBinaryDownload {
     return extractDir;
   }
 
+  async extractTarGz(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    const extract = tar.extract();
+    extract.on('entry', (header, stream, next) => {
+      if (filter(header.name))
+        stream.pipe(
+          fs.createWriteStream(path.resolve(extractDir, path.basename(header.name)), {
+            mode: 0o775,
+          })
+        );
+
+      stream.on('end', () => next());
+      stream.resume();
+    });
+
+    return new Promise((resolve, reject) => {
+      fs.createReadStream(mongoDBArchive)
+        .pipe(createUnzip())
+        .pipe(extract)
+        .on('end', () => resolve())
+        .on('error', (e) => reject(e));
+    });
+  }
+
+  async extractZip(
+    mongoDBArchive: string,
+    extractDir: string,
+    filter: (file: string) => boolean
+  ): Promise<void> {
+    return new Promise((resolve, reject) => {
+      yauzl.open(mongoDBArchive, { lazyEntries: true }, (e, zipfile) => {
+        if (e || !zipfile) reject(e);
+        zipfile?.readEntry();
```

Comment:
because `zipfile` is existing please remove `?` (and if typescript says it dosnt, surround `reject` with "{}" and add an `return`

Reply:
Is it ok to leave it on one line with a return, or do you prefer it with "{}" in this case?

Reply:
from what i see it is project standard to have it surrounded, the only exception i found is when options are set-up ([example](https://github.com/nodkz/mongodb-memory-server/blob/master/packages/mongodb-memory-server-core/src/MongoMemoryReplSet.ts#L126-L129))

so let it be / change it if you think so, and i will open an issue about this to make it standart

Reply:
sorry, my (now hidden) comment has on the wrong comment
---
> Is it ok to leave it on one line with a return, or do you prefer it with "{}" in this case?

in this case it is neccessary because you cant have `reject(e)` and `return` on one line, and to let typescript know that (in this case) `zipfile` is not `null/undefined` you will need to also add an return statement ([and here too](https://github.com/nodkz/mongodb-memory-server/pull/274#discussion_r388978181))